### PR TITLE
opt[dnm]: tweak cost model

### PR DESCRIPTION
**Pull request by justinj**
tl;dr this work is still really messy and incomplete, and I'm not happy with the presentation, but I needed to get something out the door.

The first commit is the coster changes, the second commit is the changes to external queries.

This PR tweaks the optimizer cost model. It is a bit incomplete:

* I haven't gone through the computations and made them understandable (like factoring out constants)
* Some things remain unchanged, since changing the scans and lookup joins brought a lot of the changes (since many operators largely piggyback off of scans for the bulk of their costing)
* I need to confirm that the cost model that I actually re-implemented in the optimizer was correct by fetching the estimated cost directly, so until then there might be bugs in my translation

This is just to let us take a look through the changed test cases that have changed to see if anything seems egregiously bad.

I can share more detailed data, but the scatterplots for the old/new costers should look roughly like this:
image

I'm a little concerned because it looks like there are some places where we're choosing to sort then merge-join. I need to figure out how to extract the optimizer's estimated costs directly to be able to debug this better, though. Will work on that.

Will post a second PR in the private repo showing the changes to customer queries, I'm having some trouble deciphering some of the changes to them.

### CODE REVIEW

**Review by RaduBerinde — COMMENTED** [review]
This is really cool work! Could you go over the methodology you used to come up with the constants after you had the data in the first plot?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/35282#-:-LZq1jXC46ZLzkZT2LXI:b-fxour6)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @andy-kimball, @RaduBerinde, and @rytaft)

**Comment by justinj** [comment]
Closing this in favour of a smaller change.
